### PR TITLE
docs: update contribution guide

### PR DESCRIPTION
**Pull request by tamtamchik**
- Hopefully, it makes the contributing guide more coherent.
- Updated dependencies

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -8,46 +8,49 @@ improving the documentation, your contribution is valuable and your effort to ma
...
-  vulnerabilities, please refer to [Bug Bounty](/README.md#bug-bounty)_.
+Issues are a great way to contribute to the project by reporting bugs or suggesting enhancements.
 
+- **Bug Reports**: If you encounter a bug, please report it using the GitHub issues feature. Before submitting, check if
+  the bug has already been reported. If it has, you can contribute by adding more details to the existing report. This
+  only applies to off-chain code (tests, scripts, etc.). For on-chain code, please refer to
+  the [Bug Bounty](/README.md#bug-bounty) program.
 - **Feature Requests**: Have an idea for a new feature or an improvement to an existing one? Submit a feature request
-  through the GitHub issues, detailing your proposed enhancements and how they would benefit the Lido Finance Core.
+  through GitHub issues, detailing your proposed enhancements and how they would benefit the Lido Finance Core.
```
Comment (TheDZhon) [inline CONTRIBUTING.md:21]:
```suggestion
  through GitHub issues, detailing your proposed enhancements and how they would benefit the Lido protocol middleware.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,46 +8,49 @@ improving the documentation, your contribution is valuable and your effort to ma
...
 
-Good documentation is crucial for any project. If you have suggestions for improving the documentation, or if you've
-noticed an omission or error, making these corrections is a significant contribution. Whether it's a typo, additional
-examples, or clearer explanations, your help in making the documentation more accessible and understandable is highly
-appreciated.
+Good documentation is crucial for any project. If you notice areas where the documentation can be improved, whether it's
+fixing typos, adding examples, or clarifying explanations, your contributions are highly valued. These improvements help
+make the project more accessible and understandable for everyone.
+
+For more extensive documentation, visit the [Lido Docs repository](https://github.com/lidofinance/docs).
```
Comment (TheDZhon) [inline CONTRIBUTING.md:29]:
```suggestion
For more extensive documentation, visit the [Lido Docs repository](https://github.com/lidofinance/docs) deployed at [docs.lido.fi](https://docs.lido.fi).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -8,46 +8,49 @@ improving the documentation, your contribution is valuable and your effort to ma
...
-for something that interests you or matches your expertise. Don't hesitate to ask for more information or clarification
-if needed before starting. If you're interested in improving tooling and CI in this repository, consider opening a
-feature request issue first to discuss it with the community of contributors.
+If you're interested in improving tooling and CI in this repository, consider opening a feature request issue first to
+discuss your ideas with the community of contributors.
 
-If you have a bigger idea on how to improve the protocol, consider publishing your proposal
-to [Lido Forum](https://research.lido.fi/).
+For larger ideas on how to improve the protocol, consider publishing your proposal in
+the [Lido Forum](https://research.lido.fi/).
```
Comment (TheDZhon) [inline CONTRIBUTING.md:41]:
```suggestion
the [Lido Research Forum](https://research.lido.fi/).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -64,87 +67,199 @@ to [Lido Forum](https://research.lido.fi/).
...
-This repository features a Hardhat-Foundry dual setup:
+This is the main development branch. All pull requests to `master` should be submitted to `develop` first for peer
+review.
 
-- Hardhat gives much more flexibility when writing complex tests;
-- The Foundry's fuzzing capabilities enable better coverage of edge cases.
+## Local Deployment
 
-#### Hardhat
+WIP
```
Comment (TheDZhon) [inline CONTRIBUTING.md:243]: Why WIP? hadn't we had a scratch deploy?

File: docs/scratch-deploy.md [hunk 5]
```diff
@@ -1,124 +1,130 @@
-# Deploy Lido protocol from scratch
+# Deploy Lido Protocol from Scratch
```
Comment (TheDZhon) [inline docs/scratch-deploy.md:1]:
```suggestion
# Deploy Lido protocol middleware from scratch
```

File: docs/scratch-deploy.md [hunk 6]
```diff
@@ -1,124 +1,130 @@
...
 
-## Disclaimer: WIP!
+> [!NOTE]
+> Work in progress. The document is a draft and may contain inaccuracies.
```
Comment (TheDZhon) [inline docs/scratch-deploy.md:4]: ?

**Review by TheDZhon — CHANGES_REQUESTED** [review]
So far so good :clap: 

Left a few suggestions to consider

File: CONTRIBUTING.md (hunk 4 above)
Comment (tamtamchik) [reply CONTRIBUTING.md:243]: Yep, I forgot to add a link. Nice catch!

File: docs/scratch-deploy.md (hunk 6 above)
Comment (tamtamchik) [reply docs/scratch-deploy.md:4]: I replaced "Disclaimer: WIP!" but you are right; it looks outdated, and I will remove it.

Comment (tamtamchik) [reply docs/scratch-deploy.md]: Fixed in e20a363

File: CONTRIBUTING.md (hunk 4 above)
Comment (tamtamchik) [reply CONTRIBUTING.md]: Fixed in e20a363

**Review by TheDZhon — APPROVED** [review]
GG :saluting_face:
